keras_hub/src/models/gemma/gemma_attention.py

```diff
@@ -118,28 +128,24 @@ def _compute_attention(
             query_normalization = 1 / np.sqrt(
                 self.hidden_dim // self.num_query_heads
             )
-        use_dot_product_attention = not (
-            self.dropout > 0.0 or (len(q.shape) != 4)
-        )
-        if has_flash_attention_support() and use_dot_product_attention:
-            if self.dropout > 0.0:
-                raise ValueError(
-                    "Flash attention does not support dropout. "
-                    "Please set `dropout` to 0.0."
-                )
+
             if attention_mask is not None:
                 attention_mask = ops.expand_dims(attention_mask, axis=1)
                 attention_mask = ops.cast(attention_mask, dtype="bool")
-
-            attention_output = ops.dot_product_attention(
-                query=q,
-                key=k,
-                value=v,
-                mask=attention_mask,
-                scale=query_normalization,
-            )
-            return attention_output
-
+            if self._can_use_flash_attention(q):
+                try:
+                    attention_output = ops.dot_product_attention(
+                        query=q,
+                        key=k,
+                        value=v,
+                        mask=attention_mask,
+                        scale=query_normalization,
+                        attn_logits_soft_cap=self.logit_soft_cap,
+                    )
+                    return attention_output
+                except Exception:
```

Comment:
let's avoid this, it could eat helpful errors in the future. try using inspect?

Reply:
done